### PR TITLE
Username and password validation

### PR DESCRIPTION
**Pull request by pmarsh-scottlogic**
Created a custom validation annotation for passwords. Wrote tests to check that bad usernames and passwords are caught

### CODE REVIEW

**Comment by pmarsh-scottlogic** [comment]
not certain what's an effective number of tests to write for the password validator...

**Comment by pmarsh-scottlogic** [comment]
this will close issue #17

File: src/main/java/com/example/demo/security/authInfo/AuthRequest.java [hunk 1]
```diff
@@ -1,18 +1,19 @@
...
+import com.example.demo.security.authInfo.passwordValidation.PasswordValidation;
 import lombok.AllArgsConstructor;
 import lombok.Getter;
 import lombok.Setter;
 import org.hibernate.validator.constraints.Length;
 
 @Getter @Setter @AllArgsConstructor
 public class AuthRequest { // POJO
-    @NotNull @Length(min = 1, max = 50)
+    @NotNull @Length(min = 1, max = 20)
```
Comment (nataliadombrowska) [inline src/main/java/com/example/demo/security/authInfo/AuthRequest.java:14]: Nice, I'm glad this is validated 👍

File: src/main/java/com/example/demo/security/authInfo/AuthRequest.java [hunk 2]
```diff
@@ -1,18 +1,19 @@
...
     private String username;
 
-    @NotNull @Length(min = 1, max = 50)
+    @PasswordValidation
```
Comment (nataliadombrowska) [inline src/main/java/com/example/demo/security/authInfo/AuthRequest.java:17]: That's a really handy annotation 😓

File: src/test/java/com/example/demo/WebAndSecurity/LoginValidationTest.java [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+                        MockMvcRequestBuilders.post("/auth/login")
+                                .contentType(MediaType.APPLICATION_JSON)
+                                .content(requestBody))
+                .andReturn();
+        assertThat(result.getResponse().getStatus()).isEqualTo(HttpStatus.BAD_REQUEST.value());
+        assertThat(result.getResponse().getContentAsString()).contains("Password should be");
+        assertThat(result.getResponse().getContentAsString()).contains("password");
+    }
+
+    // etc etc
```
Comment (nataliadombrowska) [inline src/test/java/com/example/demo/WebAndSecurity/LoginValidationTest.java:124]: Nitpick - maybe delete this comment just so the code on main is a bit cleaner 👍

**Review by nataliadombrowska — APPROVED** [review]
Nice!

File: src/test/java/com/example/demo/WebAndSecurity/LoginValidationTest.java (hunk 3 above)
Comment (pmarsh-scottlogic) [reply src/test/java/com/example/demo/WebAndSecurity/LoginValidationTest.java:124]: this is to indicate that for an actual production bit of code I'd do a bunch more.

File: src/main/java/com/example/demo/security/authInfo/AuthRequest.java (hunk 2 above)
Comment (pmarsh-scottlogic) [reply src/main/java/com/example/demo/security/authInfo/AuthRequest.java:17]: pretty good, eh?

**Pull request merged**